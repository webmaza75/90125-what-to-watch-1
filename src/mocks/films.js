const filmList = [
  {
    backgroundColor: `#9B7E61`,
    backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    director: `Steven Spielberg`,
    genre: `Adventure`,
    id: 1,
    isFavorite: false,
    title: `War of the Worlds`,
    posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg`,
    picture: `https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: 6.5,
    ratingLevel: `Good`,
    released: 2005,
    runTime: 116,
    scoresCount: 386834,
    starring: [
      `Tom Cruise`,
      `Dakota Fanning`,
      `Tim Robbins`,
    ],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
  },
  {
    backgroundColor: `#F1E9CE`,
    backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/Macbeth.jpg`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    director: `Justin Kurzel`,
    genre: `Drama`,
    id: 2,
    isFavorite: false,
    title: `Macbeth`,
    posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
    picture: `https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 6.6,
    ratingLevel: `Good`,
    released: 2015,
    runTime: 113,
    scoresCount: 48798,
    starring: [
      `Michael Fassbender`,
      `Marion Cotillard`,
      `Jack Madigan`
    ],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
  },
  {
    backgroundColor: `#D8E3E5`,
    backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg`,
    description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
    director: `Wes Anderson`,
    genre: `Adventure`,
    id: 3,
    isFavorite: false,
    title: `Moonrise Kingdom`,
    posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg`,
    picture: `https://es31-server.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 7.8,
    ratingLevel: `Good`,
    released: 2012,
    runTime: 94,
    scoresCount: 291183,
    starring: [
      `Jared Gilman`,
      `Kara Hayward`,
      `Bruce Willis`
    ],
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`
  }
];

export default filmList;
