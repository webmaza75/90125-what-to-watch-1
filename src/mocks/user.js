const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

export default user;
