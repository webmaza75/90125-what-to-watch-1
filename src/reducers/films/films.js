import ActionTypes from '../../actions/action-types.js';
import {ALL_GENRES} from '../../consts.js';
import {ActionCreator} from '../../actions/actions.js';
import {sortComments} from '../../utils.js';
import {getMovieRatingLevel} from '../../utils.js';

const initialState = {
  films: [],
  filter: ALL_GENRES,
  comments: []
};

export const transform = (data) => ({
  backgroundColor: data.background_color,
  backgroundImage: data.background_image,
  description: data.description,
  director: data.director,
  genre: data.genre,
  id: data.id,
  isFavorite: data.is_favorite,
  title: data.name,
  posterImage: data.poster_image,
  picture: data.preview_image,
  src: data.preview_video_link,
  rating: data.rating,
  ratingLevel: getMovieRatingLevel(data.rating),
  released: data.released,
  runTime: data.run_time,
  scoresCount: data.scores_count,
  starring: data.starring,
  videoLink: data.video_link
});

const Operation = {
  loadFilms: () => (dispatch, getState, api) => {
    return api.get(`/films`)
    .then((response) => {
      const result = response.data.map(transform);
      dispatch(ActionCreator.loadFilms(result));
    });
  },
  loadComments: (filmId) => (dispatch, getState, api) => {
    return api.get(`/comments/${filmId}`)
    .then((response) => {
      const result = response.data.sort(sortComments);
      dispatch(ActionCreator.loadComments(result));
    });
  },
  addComment: (id, params) => (dispatch, getState, api) => {
    return api
      .post(`/comments/${id}`, params)
      .then((res) => {
        dispatch(ActionCreator.addComment(res.data));
      }).catch((error) => {
        dispatch(ActionCreator.submitCommentsError(error.response.data.error));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.LOAD_FILMS:
      return ({
        ...state,
        films: action.payload,
      });
    case ActionTypes.CHANGE_GENRE:
      return {
        ...state,
        filter: action.payload
      };
    case ActionTypes.LOAD_COMMENTS:
      return ({
        ...state,
        comments: action.payload,
      });
    case ActionTypes.ADD_COMMENT:
      return ({
        ...state,
        comments: action.payload,
      });
  }
  return state;
};

export {
  Operation,
  reducer
};
