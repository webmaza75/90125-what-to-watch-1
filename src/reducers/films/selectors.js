import {createSelector} from 'reselect';
import NameSpace from '../name-spaces.js';
import {ALL_GENRES} from '../../consts.js';

const NAME_SPACE = NameSpace.FILMS;

export const getFilms = (state) => {
  return state[NAME_SPACE].films;
};

export const getActiveFilter = (state) => {
  return state[NAME_SPACE].filter;
};

export const getFilmsByGenre = createSelector(
    getFilms,
    getActiveFilter,
    (films, filter) => {
      if (filter === ALL_GENRES) {
        return films;
      }
      return films.filter((item) => item.genre === filter);
    }
);

export const getGenres = createSelector(
    getFilms,
    (films) => {
      const allGenres = films.map(({genre}) => genre);
      return [ALL_GENRES, ...new Set(allGenres)];
    }
);

export const getFavoriteList = createSelector(
    getFilms,
    (films) => {
      return films.filter((item) => item.isFavorite === true);
    }
);

export const getMovieById = (state, filmId) => {
  return state[NAME_SPACE].films.find((item) => item.id === +filmId);
};

export const getComments = (state) => {
  return state[NAME_SPACE].comments;
};
