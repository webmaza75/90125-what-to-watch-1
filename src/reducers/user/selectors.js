import NameSpace from '../name-spaces.js';

const NAME_SPACE = NameSpace.USER;

export const getUser = (state) => {
  return state[NAME_SPACE].userInfo;
};

export const getError = (state) => {
  return state[NAME_SPACE].error;
};

export const getValidationError = (state) => {
  return state[NAME_SPACE].validationError;
};

export const checkIsAuthorizedUser = (state) => {
  const user = state[NAME_SPACE].userInfo;
  return !!(user && user.id);
};

export const checkIsRequiredAuthorization = (state) => {
  return state[NAME_SPACE].isRequiredAuthorization;
};

export const checkIsLoadedUserInfo = (state) => {
  return state[NAME_SPACE].isLoadedUserInfo;
};
