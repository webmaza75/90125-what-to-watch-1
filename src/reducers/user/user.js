import ActionType from '../../actions/action-type.js';
import {ActionCreator} from '../../actions/actions.js';

const initialState = {
  userInfo: undefined,
  error: undefined,
  validationError: undefined
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SIGN_IN_USER:
      return {
        ...state,
        userInfo: action.payload
      };
    case ActionType.SIGN_IN_USER_ERROR:
      return {
        ...state,
        error: action.payload
      };
    case ActionType.VALIDATE_SIGN_IN_USER_ERROR:
      return {
        ...state,
        validationError: action.payload
      };
    case ActionType.RESET_ERRORS:
      return {
        ...state,
        validationError: action.payload
      };
    case ActionType.LOGOUT_USER:
      return initialState;
  }
  return state;
};

const transformUserData = (data) => {
  const {avatar_url: avatarUrl, ...other} = data;
  return {avatarUrl, ...other};
};

const Operation = {
  signInUser: (params) => (dispatch, getState, api) => {
    return api
      .post(`/login`, params)
      .then((res) => {
        const userInfo = transformUserData(res.data);
        dispatch(ActionCreator.signInUser(userInfo));
      }).catch((error) => {
        dispatch(ActionCreator.signInUserError(error.message));
      });
  },
  checkUser: () => (dispatch, getState, api) => {
    return api
      .get(`/login`)
      .then((res) => {
        const userInfo = transformUserData(res.data);
        dispatch(ActionCreator.signInUser(userInfo));
      }).catch((error) => {
        dispatch(ActionCreator.logoutUser());
        dispatch(ActionCreator.signInUserError(error.message));
      });
  }
};

export {
  Operation,
  reducer,
  transformUserData
};
