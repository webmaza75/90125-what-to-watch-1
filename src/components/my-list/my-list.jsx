import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {itemShape} from '../../models.js';
import FilmList from '../film-list/film-list.jsx';
import {getFavoriteList} from '../../reducers/films/selectors.js';
import {Operation} from '../../reducers/films/films.js';
import {getUser} from '../../reducers/user/selectors.js';
import Footer from '../footer/footer.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.js';
import MyListHeader from '../my-list-header/my-list-header.jsx';
import {userInfo} from '../../models.js';

const FilmListWrapped = withActiveItem(FilmList);

class MyList extends PureComponent {
  componentDidMount() {
    const {onGetFavoriteList} = this.props;
    onGetFavoriteList();
  }

  render() {
    const {
      myList,
      user
    } = this.props;

    return <div className="user-page">
      <MyListHeader user={user} />

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <FilmListWrapped films={myList} />
        <Footer />
      </section>
    </div>;
  }
}

MyList.propTypes = {
  user: userInfo,
  myList: PropTypes.arrayOf(itemShape),
  onGetFavoriteList: PropTypes.func
};

const mapStateToProps = (state) => ({
  user: getUser(state),
  myList: getFavoriteList(state)
});

const mapDispatchToProps = {
  onGetFavoriteList: Operation.loadFavorites
};

export {MyList};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MyList);
