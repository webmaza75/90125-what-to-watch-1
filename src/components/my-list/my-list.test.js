import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import {MyList} from './my-list.jsx';
import films from '../../mocks/films.js';
import user from '../../mocks/user.js';

it(`MyList correctly renders without any errors`, () => {
  const tree = renderer
    .create(<MemoryRouter initialEntries={[`/`]}>
      <MyList
        myList={films}
        user={user}
      />
    </MemoryRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
