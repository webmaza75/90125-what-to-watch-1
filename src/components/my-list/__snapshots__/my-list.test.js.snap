// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList correctly renders without any errors 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="https://es31-server.appspot.comimg/1.png"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <a
          href="/film/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="War of the Worlds"
              height="175"
              src="https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/1"
            onClick={[Function]}
          >
            War of the Worlds
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <a
          href="/film/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="Macbeth"
              height="175"
              src="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/2"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <a
          href="/film/3"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="Moonrise Kingdom"
              height="175"
              src="https://es31-server.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/3"
            onClick={[Function]}
          >
            Moonrise Kingdom
          </a>
        </h3>
      </article>
    </div>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </section>
</div>
`;
